src/box/tuple.c

```diff
@@ -796,6 +800,16 @@ box_tuple_validate(box_tuple_t *tuple, box_tuple_format_t *format)
 
 /* }}} box_tuple_* */
 
+box_tuple_t *
+box_tuple_new_with_validation(box_tuple_format_t *format, const char *data,
+			      const char *end, bool validate)
+{
+	struct tuple *ret = tuple_new(format, data, end, validate);
+	if (ret == NULL)
```

Comment:
```
tarantool> box.tuple.new({1, 2, 3}, {format = {{type = 'int', name = 'id'}, {type = 'int', name = 'id'}, {type = 'int', name = 'id'}}})
---
- error: Space field 'id' is duplicate
...
```

Should be tuple field?

Reply:
This error is set here: 
https://github.com/tarantool/tarantool/blob/c5f5ee0d51c5ece443f56d19222400893fa68a7f/src/box/tuple_dictionary.c#L86-L90

Not sure how we should distinguish tuple format from space format.

Reply:
Discussed with @locker: we would rather opt to wait for tarantool/roadmap-internal#226, which would solve such problems.